lib/oli_web/live/products/details/image_upload.ex

```diff
@@ -0,0 +1,128 @@
+defmodule OliWeb.Products.Details.ImageUpload do
+  use Surface.Component
+
+  alias Surface.Components.{Form, LiveFileInput}
+  alias Surface.Components.Form.{
+    Label,
+    Submit
+  }
+
+  prop product, :any, required: true
+  prop updates, :any, required: true
+  prop changeset, :any, default: nil
+  prop uploads, :map, required: true
+  prop upload_event, :event, required: true
+  prop change, :event, required: true
+  prop cancel_upload, :event, required: true
+
+  @spec render(any) :: Phoenix.LiveView.Rendered.t()
+  def render(assigns) do
+    ~F"""
+    <div class="container">
+
+      <div class="row">
+        <div class="col-12">
+          <Form for={@changeset} submit={@upload_event} change="validate_image" opts={[id: "img-upload-form"]}>
+
+          <section>
+            <div id="drag-and-drop-zone" class="drag-and-drop-zone mb-2 py-4 w-75" phx-drop-target={@uploads.cover_image.ref}>
+
+            <div class="row d-flex justify-content-center">
+                <div class="col-6 d-flex justify-content-center">
+
+                  <div class="form-group input-file-form-group">
+                    <LiveFileInput upload={@uploads.cover_image} class="img-input-file" opts={[disaled: @uploads.cover_image.entries != []]}/>
```

Comment:
Thanks for catching this, I'm actually removing it because it makes more sense that people can reselect after having uploaded something, without having to click on the cancel button to re-enable the input.